database/migrations/2020_01_12_000000_create_cart_table.php

```diff
@@ -1,30 +1,31 @@
 <?php
 
-use Illuminate\Support\Facades\Schema;
-use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
 
-class CreateShoppingcartTable extends Migration
+class CreateCartTable extends Migration
 {
     /**
      * Run the migrations.
      */
     public function up()
     {
-        Schema::create('shoppingcart', function (Blueprint $table) {
+        Schema::create('carts', function (Blueprint $table) {
+            $table->bigIncrements('id');
             $table->string('identifier');
             $table->string('instance');
             $table->longText('content');
             $table->nullableTimestamps();
 
-            $table->primary(['identifier', 'instance']);
+            $table->unique(['identifier', 'instance']);
```

Comment:
Ahh, yes. I added `$table->bigIncrements('id')` to this table, in which case `primary(...)` had to be changed to `unique(...)`, because the table can only have one primary key (obviously).